File: Realidad Virtual/1 Primer Momento/Tarea 1/main.cpp
#include <iostream>
#include <list>
#include <string>

using namespace std;

class Usuario {
public:
    int id;
    string nombre;
    string usuario;

    Usuario(int id, string nombre, string usuario) : id(id), nombre(nombre), usuario(usuario) {}
};

class CRUD {
private:
    list<Usuario> usuarios;

public:
    void crearUsuario(int id, string nombre, string usuario) {
        usuarios.push_back(Usuario(id, nombre, usuario));
    }

    void leerUsuarios() {
        for (const auto& usuario : usuarios) {
            cout << "ID: " << usuario.id << ", Nombre: " << usuario.nombre << ", Usuario: " << usuario.usuario << endl;
        }
    }

    void actualizarUsuario(int id, string nombre, string usuario) {
        for (auto& usr : usuarios) {
            if (usr.id == id) {
                usr.nombre = nombre;
                usr.usuario = usuario;
                return;
            }
        }
        cout << "Usuario con ID " << id << " no encontrado." << endl;
    }

    void eliminarUsuario(int id) {
        for (auto it = usuarios.begin(); it != usuarios.end(); ++it) {
            if (it->id == id) {
                usuarios.erase(it);
                return;
            }
        }
        cout << "Usuario con ID " << id << " no encontrado." << endl;
    }
};

int main() {
    CRUD crud;

    int opcion;
    while (true) {
        cout << "\n1. Crear usuario\n2. Leer usuarios\n3. Actualizar usuario\n4. Eliminar usuario\n5. Salir\n";
        cout << "Selecciona una opción: ";
        cin >> opcion;

        if (opcion == 5) break;

        int id;
        string nombre, usuario;

        switch (opcion) {
            case 1:
                cout << "Ingrese ID: ";
                cin >> id;
                cout << "Ingrese nombre: ";
                cin >> nombre;
                cout << "Ingrese usuario: ";
                cin >> usuario;
                crud.crearUsuario(id, nombre, usuario);
                break;
            case 2:
                crud.leerUsuarios();
                break;
            case 3:
                cout << "Ingrese ID del usuario a actualizar: ";
                cin >> id;
                cout << "Ingrese nuevo nombre: ";
                cin >> nombre;
                cout << "Ingrese nuevo usuario: ";
                cin >> usuario;
                crud.actualizarUsuario(id, nombre, usuario);
                break;
            case 4:
                cout << "Ingrese ID del usuario a eliminar: ";
                cin >> id;
                crud.eliminarUsuario(id);
                break;
            default:
                cout << "Opción no válida." << endl;
                break;
        }
    }

    return 0;
}

File: Realidad Virtual/1 Primer Momento/Tarea 1/Main.cs
static void Main(string[] args)
    {
        var manager = new PersonajeManager();

        while (true)
        {
            Console.WriteLine("\nGestión de Personajes:");
            Console.WriteLine("1. Crear Personaje");
            Console.WriteLine("2. Listar Personajes");
            Console.WriteLine("3. Actualizar Personaje");
            Console.WriteLine("4. Eliminar Personaje");
            Console.WriteLine("5. Salir");
            Console.Write("Seleccione una opción: ");
            var opcion = Console.ReadLine();

            switch (opcion)
            {
                case "1":
                    Console.Write("Nombre: ");
                    var nombre = Console.ReadLine();
                    Console.Write("Descripcion: ");
                    var descripcion = Console.ReadLine();
                    Console.Write("Tipo: ");
                    var tipo = Console.ReadLine();
                    Console.WriteLine("Ingrese las acciones (separadas por comas): ");
                    var accionesInput = Console.ReadLine();
                    var acciones = accionesInput.Split(',').Select(a => a.Trim()).ToList();
                    manager.CrearPersonaje(nombre, descripcion, tipo, acciones);
                    break;
                case "2":
                    manager.ListarPersonajes();
                    break;
                case "3":
                    Console.Write("ID del Personaje a actualizar: ");
                    var idActualizar = int.Parse(Console.ReadLine());
                    Console.Write("Nuevo Nombre: ");
                    nombre = Console.ReadLine();
                    Console.Write("Nueva Descripcion: ");
                    descripcion = Console.ReadLine();
                    Console.Write("Nuevo Tipo: ");
                    tipo = Console.ReadLine();
                    Console.WriteLine("Ingrese las nuevas acciones (separadas por comas): ");
                    accionesInput = Console.ReadLine();
                    acciones = accionesInput.Split(',').Select(a => a.Trim()).ToList();
                    manager.ActualizarPersonaje(idActualizar, nombre, descripcion, tipo, acciones);
                    break;
                case "4":
                    Console.Write("ID del Personaje a eliminar: ");
                    var idEliminar = int.Parse(Console.ReadLine());
                    manager.EliminarPersonaje(idEliminar);
                    break;
                case "5":
                    return;
                default:
                    Console.WriteLine("Opción no válida.");
                    break;
            }
        }
    }